Created CONTRIBUTING.md (#3)
idvogados/backend: CONTRIBUTING.md

# Contribuindo com o projeto

Ficamos felizes que você quer contribuir com o projeto!
Nesse documento estão listados os pré-requisitos que você precisa entender antes de contribuir com o projeto.

## Código de Conduta

É necessário que você leia e entenda o código de conduta antes de contribuir.
Para acessá-lo, clique [aqui](https://github.com/idvogados/code-of-conduct)

## Como contribuir

> Para contribuir com o projeto é necessário identificar issues que ainda estão abertas para aí sim abrir um PR. O passo a passo de como forkar o repositório e como abrir um PR seguem abaixo.

Reviewer: Acredito que o termo forkar não é muito bom, eis minha sugestão.
```suggestion
Para contribuir com o projeto é necessário identificar issues que ainda estão abertas para aí sim abrir um PR. O passo a passo de como fazer um *fork* o repositório e como abrir um PR seguem abaixo.
```

Author: Pensei exatamente nisso na hora que estava escrevendo, mas preferi deixar assim e descobrir as opiniões de outras pessoas. :)

> O projeto hoje está dividido em 2 repositórios, o [**frontend**](https://github.com/idvogados/frontend) e o [**backend**](https://github.com/idvogados/backend) os passos a passos abaixo são idênticos para os dois repositórios.

Reviewer 2: Cada projeto terá seu próprio contributing. 

Pode ser que os requisitos para contribuir com o backend seja diferente do frontend e assim por diante.

Remove essa parte mencionando o projeto em 2 repositórios e que os passos serão identicos (mesmo que por enquanto iremos duplicar esse arquivo no outro repo)

> O projeto hoje está dividido em 2 repositórios, o [**frontend**](https://github.com/idvogados/frontend) e o [**backend**](https://github.com/idvogados/backend) os passos a passos abaixo são idênticos para os dois repositórios.

Reviewer 2: Cada projeto terá seu próprio contributing.

Pode ser que os requisitos para contribuir com o backend seja diferente do frontend e assim por diante.

Acho melhor remover essa parte (mesmo que por enquanto iremos duplicar esse arquivo nos outros repos)

```suggestion

```

Author: Ótimo, como disse, mesmo sem muitas especificações preferi mandar o pr para poder já começar a participar! :)

Author: E faz sentido ter especificações de contribuição diferentes, concordo com a decisão!

> 1. ### **Como forkar o Repositório**

Reviewer: Como essa linha já possui as tags #, não precisa dos asteriscos.
```suggestion
1. ### Como forkar o Repositório
```

Author: Justo

   1- Esteja no repositório que deseja contribuir no Github, e no canto superior direito há um botão **_Fork_**, clique nele.

>    2- Assim que o processo de fork for finalizado você será redirecionado para o seu repositório que foi forkado a partir do idvogados.

Reviewer: ```suggestion
   2- Assim que o processo de fork for finalizado você será redirecionado para a sua cópia do repositório a partir do idvogados.
```

2. ### **Como criar um PR**

>    1- Inicialmente é necessário fazer as alterações de código dentro do seu repositório forkado localmente, em uma nova branch.

Reviewer: Acho que podemos remover o termo forkado aqui tbm, o que vc acha?
```suggestion
   1- Inicialmente é necessário fazer as alterações de código dentro do seu repositório localmente, em uma nova branch.
```

Author: Acho uma ótima

Reviewer: Só apertar no botão `Commit suggestion` para fazer o commit das alterações que vc concordar. (;

   2- Após ter comitado todas as alterações, subir sua branch para o repositório remoto forkado.

   3- Depois disso ir ao repositório original, Clicar em **New Pull Request**

>    4- Na página de comparação(Compare Changes), selecionar **compare across forks**.

Reviewer: Como é um termo em inglês sugiro que fique em itálico.
```suggestion
   4- Na página de comparação(_Compare Changes_), selecionar **compare across forks**.
```

>    5- Em **base fork** selecione o repositório e a branch **_para a qual você deseja subir as alterações_**

Reviewer 2: Todo PR deverá ser aberto para a branch padrão do repositório (no caso a branch `dev`)

>    5- Em **base fork** selecione o repositório e a branch **_para a qual você deseja subir as alterações_**

Reviewer 2: Irá ser decidido ainda. Com relação à outros procedimentos/restrições decididos pelos tech leads.

Mas por enquanto todo PR deverá ser aberto para a branch padrão do repositório (no caso a branch dev)
```suggestion
   5- Em **base fork** selecione o repositório e a branch **_dev_**
```

Author: Ótimo! Deixa comigo que vou alterar

   6- Em **head fork** selecione o repositório e a branch **_nas quais você aplicou as alterações_**

>    7- Adicione uma descrição e um Título para o seu PR

Reviewer 3: Acredito que aqui deve recomendar para usar o template de [pull request](https://github.com/idvogados/backend/blob/dev/.github/PULL_REQUEST_TEMPLATE.md) que foi criado para o projeto.
```suggestion
   7- Preencha os campos necessários para criar uma PR, seguindo modelo do template.
```

O que acha ?

Author: Acho uma ótima! Mas como ao criar o PR o modelo já está inserido no campo de descrição, podemos só instruir para preenchê-lo corretamente!

Reviewer 3: @pedrocecchetti maravilha, acho uma boa.
